internal/internal_workflow.go

```diff
@@ -216,6 +216,7 @@ type (
 		WaitForCancellation      bool
 		WorkflowIDReusePolicy    enumspb.WorkflowIdReusePolicy
 		WorkflowIDConflictPolicy enumspb.WorkflowIdConflictPolicy
+		OnConflictOptions        *OnConflictOptions
```

Comment:
I'm adding this `WorkflowOptions`, but not sure if I should. I need because the Nexus handler workflow is executed in the test framework as a child workflow, and I need this to resolve workflow ID conflict, ie., starting the same operation multiple times.

I can see this is used in certain operations. For example, I saw `WorkflowOptions` is built for `ExecuteChildWorkflow`, and `OnConflictOptions` is not an options for executing child workflow.

Btw, `WorkflowIDConflictPolicy` was added here a while ago, but I didn't see any usage of this field anywhere... I'm using it now here though.

cc: @Quinn-With-Two-Ns @cretz

Reply:
`WorkflowIDConflictPolicy` is not supported for child workflows https://github.com/temporalio/temporal/issues/6799 so yeah it probably shouldn't have been added here. 

Is it reasonable to refactor the Nexus handler workflow to not be a "child" in the test framework?

Reply:
Agreed, we shouldn't add an unused field here (even if there happens to be another unused field here)

Reply:
I think Rodrigo is saying it is no longer an unused field, but I guess it is unused when executing outside the test enviorment?

Reply:
Yes, `WorkflowIDConflictPolicy` is unused since it was added in https://github.com/temporalio/sdk-go/pull/1563, but I need it now with this PR.

As for not being a child workflow, I'm not sure how we would be able to run the Nexus workflow without it since the test environment only allows to run one main/root workflow at a time. cc: @bergundy

Reply:
Oh, my mistake, I thought this was user-facing child workflow options or something. If this is an `internal` only thing I have no opinion/preference, will defer to Quinn.

Reply:
`WorkflowOptions` is somewhat user facing, isn't it? The user could get it from the workflow context.
The two fields I mentioned, it's only used in the test environment. Maybe I could just make them unexported?

Reply:
How can users get `WorkflowOptions` from the workflow context?

Reply:
Well, it's not exported, but technically, the user could write directly `ctx.Value("wfEnvOptions")`.
Otherwise, it seems it's only for `internal` only.

Reply:
Yeah I think the only reason the type is even exported anyway is for the nexus test environment and maybe the PHP SDK